src/modules/feature-limit/feature-limit-migration.service.ts

```diff
@@ -0,0 +1,132 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { SequelizeUserRepository } from '../user/user.repository';
+import { SequelizeFeatureLimitsRepository } from './feature-limit.repository';
+import { AxiosError } from 'axios';
+import { User } from '../user/user.domain';
+import { PaymentsService } from 'src/externals/payments/payments.service';
+import { PLAN_FREE_TIER_ID } from './limits.enum';
+
+@Injectable()
+export class FeatureLimitsMigrationService {
+  private readonly delayBetweenUsers = 10;
+  private readonly maxRetries = 3;
+  private readonly paymentsAPIThrottlingDelay = 60000;
+
+  private planIdTierIdMap: Map<string, string>; // PlanId/tierId mapping
+
+  constructor(
+    private userRepository: SequelizeUserRepository,
+    private tiersRepository: SequelizeFeatureLimitsRepository,
+    private paymentsService: PaymentsService,
+  ) {}
+
+  async asignTiersToUsers() {
+    const limit = 20;
+    let offset = 0;
+    let processed = 0;
+    await this.loadTiers();
+
+    while (true) {
```

Comment:
Hey there is a way to do this without a true and a break. Let's use a bit of math here:

```
let resultsCount = 0;

while (resultsCount % limit === 0) {
 ....
 resultsCount = users.length;
}
```

This is dynamic with the limit and handles cases where `users.length` could be different than 0, for instance: the last page of the pagination.